apiserver/facades/agent/machineactions/machineactions.go

```diff
@@ -45,6 +47,10 @@ type OperationService interface {
 	// NOTE: This watcher will emit events for tasks changing their statuses to
 	// PENDING only.
 	WatchMachineTaskNotifications(ctx context.Context, machineName coremachine.Name) (watcher.StringsWatcher, error)
+
+	// GetMachineTaskIDsWithStatus retrieves all task IDs for a machine with the specified status.
+	// Returns an error if the operation fails.
```

Comment:
nit: please rephrase without the work `operation`. It's used in a different context already in this domain.